specs/_features/eip7594/das-core.md

```diff
@@ -0,0 +1,225 @@
+| - | - | - |
+| `NUMBER_OF_COLUMNS` | `uint64((FIELD_ELEMENTS_PER_BLOB * 2) // FIELD_ELEMENTS_PER_CELL)` (= 128) | Number of columns in the extended data matrix. |
+
+### Custody setting
+
+| Name | Value | Description |
+| - | - | - |
+| `SAMPLES_PER_SLOT` | `8` | Number of random samples a node queries per slot |
+| `CUSTODY_REQUIREMENT` | `2` | Minimum number of columns an honest node custodies and serves samples from |
+| `TARGET_NUMBER_OF_PEERS` | `70` | Suggested minimum peer count |
+
+### Helper functions
+
+#### `get_custody_lines`
+
+```python
+def get_custody_lines(node_id: NodeID, custody_size: uint64) -> Sequence[LineIndex]:
+    assert custody_size <= NUMBER_OF_COLUMNS
+    all_items = list(range(NUMBER_OF_COLUMNS))
+    line_index = node_id % NUMBER_OF_COLUMNS
+    return [LineIndex(all_items[(line_index + i) % len(all_items)]) for i in range(custody_size)]
+```
+
+#### `compute_extended_data`
+
+```python
+def compute_extended_data(data: Sequence[BLSFieldElement]) -> Sequence[BLSFieldElement]:
+    # TODO
+    # pylint: disable=unused-argument
+    ...
+```
+
+#### `compute_extended_matrix`
+
+```python
+def compute_extended_matrix(blobs: Sequence[Blob]) -> FlatExtendedMatrix:
+    matrix = [compute_extended_data(blob) for blob in blobs]
+    return FlatExtendedMatrix(matrix)
+```
+
+#### `get_data_column_sidecars`
+
+```python
+def get_data_column_sidecars(signed_block: SignedBeaconBlock,
+                             blobs: Sequence[Blob]) -> Sequence[DataColumnSidecar]:
+    signed_block_header = compute_signed_block_header(signed_block)
+    block = signed_block.message
+    kzg_commitments_inclusion_proof = compute_merkle_proof(
+        block.body,
+        get_generalized_index(BeaconBlockBody, 'blob_kzg_commitments'),
+    )
+    cells_and_proofs = [compute_cells_and_proofs(blob) for blob in blobs]
+    blob_count = len(blobs)
+    cells = [cells_and_proofs[i][0] for i in range(blob_count)]
+    proofs = [cells_and_proofs[i][1] for i in range(blob_count)]
+    sidecars = []
+    for column_index in range(NUMBER_OF_COLUMNS):
+        column = DataColumn([cells[row_index][column_index]
+                             for row_index in range(blob_count)])
+        kzg_proof_of_column = [proofs[row_index][column_index]
+                               for row_index in range(blob_count)]
+        sidecars.append(DataColumnSidecar(
+            index=column_index,
+            column=column,
+            kzg_commitments=block.body.blob_kzg_commitments,
+            kzg_proofs=kzg_proof_of_column,
+            signed_block_header=signed_block_header,
+            kzg_commitments_inclusion_proof=kzg_commitments_inclusion_proof,
+        ))
+    return sidecars
+```
+
+## Custody
+
+### Custody requirement
+
+Each node downloads and custodies a minimum of `CUSTODY_REQUIREMENT` columns per slot. The particular columns that the node is required to custody are selected pseudo-randomly (more on this below).
```

Comment:
```suggestion
Each node downloads and custodies a minimum of `CUSTODY_REQUIREMENT` consecutive columns per slot. The particular set of columns that the node is required to custody are selected pseudo-randomly (more on this below).
```

Reply:
Resolved by changing the custody assignment calculation (https://github.com/ethereum/consensus-specs/pull/3574#issuecomment-1914298172).